cpp/src/io/statistics/column_stats.cu

```diff
@@ -295,73 +213,54 @@ void __device__ gatherStringColumnStats(stats_state_s *s, uint32_t t, Storage &s
 {
   using block_reduce = cub::BlockReduce<uint32_t, block_size>;
   uint32_t len_sum   = 0;
-  const char *smin   = nullptr;
-  const char *smax   = nullptr;
-  uint32_t lmin      = 0;
-  uint32_t lmax      = 0;
   uint32_t nn_cnt    = 0;
-  bool has_minmax;
-  string_stats minval, maxval;
+  bool has_minmax    = false;
+
+  string_view minimum_value = string_view::max();
+  string_view maximum_value = string_view::min();
 
   for (uint32_t i = 0; i < s->group.num_rows; i += block_size) {
-    uint32_t r                = i + t;
-    uint32_t row              = r + s->group.start_row;
-    const uint32_t *valid_map = s->col.valid_map_base;
-    uint32_t is_valid         = (r < s->group.num_rows && row < s->col.num_values)
-                          ? (valid_map) ? (valid_map[(row + s->col.column_offset) >> 5] >>
-                                           ((row + s->col.column_offset) & 0x1f)) &
-                                            1
-                                        : 1
+    uint32_t r        = i + t;
+    uint32_t row      = r + s->group.start_row;
+    uint32_t is_valid = (r < s->group.num_rows && row < s->col.leaf_column->size())
+                          ? s->col.leaf_column->is_valid(row)
                           : 0;
     if (is_valid) {
-      const nvstrdesc_s *str_col = static_cast<const nvstrdesc_s *>(s->col.column_data_base);
-      uint32_t len               = (uint32_t)str_col[row].count;
-      const char *ptr            = str_col[row].ptr;
-      len_sum += len;
-      if (!smin || nvstr_is_lesser(ptr, len, smin, lmin)) {
-        lmin = len;
-        smin = ptr;
-      }
-      if (!smax || nvstr_is_greater(ptr, len, smax, lmax)) {
-        lmax = len;
-        smax = ptr;
-      }
+      has_minmax = true;
+      auto str   = s->col.leaf_column->element<string_view>(row);
+      len_sum += str.size_bytes();
+      if (str > maximum_value) { maximum_value = str; }
+      if (str < minimum_value) { minimum_value = str; }
     }
     nn_cnt += __syncthreads_count(is_valid);
   }
   if (!t) {
     s->ck.non_nulls  = nn_cnt;
     s->ck.null_count = s->group.num_rows - nn_cnt;
   }
-  minval = WarpReduceMinString(smin, lmin);
-  maxval = WarpReduceMaxString(smax, lmax);
+  minimum_value = cudf::strings::string::warp_reduce(minimum_value, cudf::DeviceMin());
```

Comment:
Just like the others (int and float), this too can be a block reduce now. Earlier the thing stopping it was that the comparators `nvstr_is_lesser/greater` weren't operator overloaded members of the class `nvstrdesc_s`. That and the fact that the identity was represented by nullptr data.